hw24_hamburger/js/pricer.js

```diff
@@ -0,0 +1,152 @@
+class Ingredient {
+	constructor(id, category, name, price, kkal) {
+		this.id = id;
+		this.category = category;
+		this.name = name;
+		this.price = price;
+		this.kkal = kkal;
+	}
+}
+
+class Gamburger {
+	constructor(gamburgerSize = null, insideIngredient = null) {
+		this.gamburgerSize = gamburgerSize;
+		this.insideIngredient = insideIngredient;
+		this.outerIngredients = new Set();
+	}
+
+	get isReady() {
+		return this.gamburgerSize != null && this.insideIngredient != null;
+	}
+
+	get moneySum() {
+		if (!this.isReady)
+			throw new Error("Is not ready.");
+
+		let sum = this.gamburgerSize.price;
+		sum += this.insideIngredient.price;
+		for (let item of this.outerIngredients) {
+			sum += item.price;
+		}
+		return sum;
+	}
+
+	get kkalSum() {
+		if (!this.isReady)
+			throw new Error("Is not ready.");
+
+		let sum = this.gamburgerSize.kkal;
+		sum += this.insideIngredient.kkal;
+		for (let item of this.outerIngredients) {
+			sum += item.kkal;
+		}
+		return sum;
+
+	}
+
+	get name() {
+		if (!this.isReady)
+			throw new Error("Is not ready.");
+
+		let name = this.gamburgerSize.name + " (";
+		name += this.insideIngredient.name;
+		for (let item of this.outerIngredients) {
+			name += ", " + item.name;
+		}
+		return name + ")";
+	}
+
+	static get builder() { return new GamburgerBuilder(); }
+}
+ 
+class GamburgerBuilder {
+	get isReady() { return this.gamburger.isReady; };
+
+	constructor() {
+		this.gamburger = new Gamburger();
+	}
+
+	withInside(ingredient) {
+		if (!ingredient || ingredient.category != 1) 
```

Comment:
```suggestion
		if (!ingredient || ingredient.category !== 1) 
```
it could broke your app. use strong check